docs: replace occurrences of "charmcraft.yaml" and "recipe" (#2154)
canonical/charmcraft: docs/howto/manage-12-factor-app-charms.rst

  charmcraft init --profile <profile>

> Charmcraft automatically creates a project file, ``requirements.txt`` and source code

Author: Maybe this one should be left as `charmcraft.yaml`?

Reviewer: Let's keep `charmcraft.yaml`, but I think including "project file" here would help pair them.

------------------------------------------

A charm integration can be added to your charmed 12-factor app by providing

> the integration and endpoint definition in your project file:
> 
> .. code-block:: yaml

Reviewer 2: To @medubelko's point on captions:

```suggestion
the integration and endpoint definition:

.. code-block:: yaml
    :caption: charmcraft.yaml
```

Author: I'll be going back to tackle all of these in a separate PR.

Reviewer 2: Yeah, just demonstrating how it can also reduce what we have to say in the body of the text.

Author: Ohhhh my bad, I didn't see the truncated text above the code block. That is a convenient way to keep the language nice and concise.
